File: android-starter/app/src/main/java/com/evplatform/driver/ui/components/TrustBadge.kt
package com.evplatform.driver.ui.components

import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.evplatform.driver.ui.theme.EVColor

/**
 * The Trust Engine+ wedge, surfaced everywhere a connector appears: map
 * pins, station detail, session start confirmation. Three tiers mirror the
 * web app's badge logic exactly (see web/static/app.js `trustBadgeHtml`) —
 * Guaranteed (score >= 90, no open Plug Watch report), Good, or Low trust.
 */
@Composable
fun TrustBadge(reliabilityScore: Double, guaranteed: Boolean) {
    val (label, color) = when {
        guaranteed -> "Guaranteed" to EVColor.lime
        reliabilityScore >= 75 -> "Good" to EVColor.amber
        else -> "Low trust" to EVColor.red
    }

    Row(
        verticalAlignment = Alignment.CenterVertically,
        modifier = Modifier
            .clip(CircleShape)
            .background(color.copy(alpha = 0.14f))
            .border(1.dp, color.copy(alpha = 0.6f), CircleShape)
            .padding(horizontal = 10.dp, vertical = 5.dp)
    ) {
        Box(Modifier.size(6.dp).clip(CircleShape).background(color))
        Spacer(Modifier.width(6.dp))
        Text("$label · ${reliabilityScore.toInt()}", color = color, fontSize = 11.sp, fontWeight = FontWeight.SemiBold)
    }
}

File: ios-starter/EVCharge/Components/TrustBadge.swift
import SwiftUI

/// The Trust Engine+ wedge, surfaced everywhere a connector appears: map
/// pins, station detail, session start confirmation. Three tiers mirror the
/// web app's badge logic exactly (see web/static/app.js `trustBadgeHtml`) —
/// Guaranteed (score >= 90, no open Plug Watch report), Good, or Low trust.
struct TrustBadge: View {
    let reliabilityScore: Double
    let guaranteed: Bool

    private var tier: (label: String, color: Color) {
        if guaranteed { return ("Guaranteed", EVTheme.lime) }
        if reliabilityScore >= 75 { return ("Good", EVTheme.amber) }
        return ("Low trust", EVTheme.red)
    }

    var body: some View {
        HStack(spacing: 6) {
            Circle().fill(tier.color).frame(width: 6, height: 6)
            Text("\(tier.label) · \(Int(reliabilityScore))")
                .font(EVTheme.Font.mono(11, weight: .semibold))
        }
        .padding(.horizontal, 10).padding(.vertical, 5)
        .background(tier.color.opacity(0.14))
        .overlay(Capsule().stroke(tier.color.opacity(0.6)))
        .clipShape(Capsule())
        .foregroundColor(tier.color)
    }
}

#Preview {
    VStack(spacing: 12) {
        TrustBadge(reliabilityScore: 97, guaranteed: true)
        TrustBadge(reliabilityScore: 82, guaranteed: false)
        TrustBadge(reliabilityScore: 41, guaranteed: false)
    }
    .padding()
    .background(EVTheme.bgDeep)
}
